docs/sdks/typescript/index.md

```diff
@@ -55,927 +67,740 @@ Each SpacetimeDB client depends on some bindings specific to your module. Create
+```typescript
+DbConnectionBuilder.withModuleName(name_or_identity: string): DbConnectionBuilder
 
-```ts
-registerReducers(...reducerClasses: ReducerClass[])
 ```
 
-#### Parameters
+Configure the SpacetimeDB domain name or `Identity` of the remote module which identifies it within the SpacetimeDB instance or cluster.
 
-| Name             | Type           | Description                   |
-| :--------------- | :------------- | :---------------------------- |
-| `reducerClasses` | `ReducerClass` | A list of classes to register |
+#### Callback `onConnect`
 
-#### Example
-
-```ts
-import SayHelloReducer from './types/say_hello_reducer';
-import AddReducer from './types/add_reducer';
-
-SpacetimeDBClient.registerReducers(SayHelloReducer, AddReducer);
+```typescript
+DbConnectionBuilder.onConnect(
+    callback: (ctx: DbConnection, identity: Identity, token: string) => void
+): DbConnectionBuilder
 ```
 
----
+Chain a call to `.onConnect(callback)` to your builder to register a callback to run when your new `DbConnection` successfully initiates its connection to the remote module. The callback accepts three arguments: a reference to the `DbConnection`, the `Identity` by which SpacetimeDB identifies this connection, and a private access token which can be saved and later passed to [`withToken`](#method-withtoken) to authenticate the same user in future connections.
 
-### `SpacetimeDBClient.registerTables`
+This interface may change in an upcoming release as we rework SpacetimeDB's authentication model.
 
-Registers table classes for use with a SpacetimeDBClient
+#### Callback `onConnectError`
 
-```ts
-registerTables(...reducerClasses: TableClass[])
+```typescript
+DbConnectionBuilder.onConnectError(
+    callback: (ctx: ErrorContext, error: Error) => void
+): DbConnectionBuilder
 ```
 
-#### Parameters
+Chain a call to `.onConnectError(callback)` to your builder to register a callback to run when your connection fails.
 
-| Name           | Type         | Description                   |
-| :------------- | :----------- | :---------------------------- |
-| `tableClasses` | `TableClass` | A list of classes to register |
+#### Callback `onDisconnect`
 
-#### Example
-
-```ts
-import User from './types/user';
-import Player from './types/player';
+```typescript
+DbConnectionBuilder.onDisconnect(
+    callback: (ctx: ErrorContext, error: Error | null) => void
+): DbConnectionBuilder
 
-SpacetimeDBClient.registerTables(User, Player);
 ```
 
----
-
-## Properties
+Chain a call to `.onDisconnect(callback)` to your builder to register a callback to run when your `DbConnection` disconnects from the remote module, either as a result of a call to [`disconnect`](#method-disconnect) or due to an error.
 
-### `SpacetimeDBClient` identity
+#### Method `withToken`
 
-The user's public [Identity](#class-identity).
-
-```
-identity: Identity | undefined
+```typescript
+DbConnectionBuilder.withToken(token: string | null): DbConnectionBuilder
```

Comment:
```suggestion
DbConnectionBuilder.withToken(token: string | undefined): DbConnectionBuilder
```

JavaScript is a silly language. This is a meaningful distinction.

Reply:
I checked the SDK code when I wrote this, and was pretty sure it said `undefined`, not `null`. But now I'm looking again, and it seems to have been changed to (or always was) `token?: string`.

That said, based on [the TypeScript handbook](https://www.typescriptlang.org/docs/handbook/2/functions.html#optional-parameters), it looks like `token?: string` means `string | undefined`, *not* `string | null`.

Am I missing something here?

Reply:
I also checked. `token?: string` does mean `string | undefined`. In your docs it's `string | null` which is not correct.

Reply:
Ah, I see. Will fix. I'm going to just write `token?: string`.